### PR TITLE
localize form items & licenses on client side

### LINKED ISSUE #492: application form item titles are not localized
from #481

### PR DESCRIPTION
**Pull request by opqdonut**
closes #492

### CODE REVIEW

**Comment by opqdonut** [comment]
TODO: do something about validation message localization

File: src/clj/rems/db/applications.clj [hunk 1]
```diff
@@ -279,10 +279,13 @@
      :value \"filled value or nil\"}"
   [application-id form-id item]
   {:id (:id item)
-   :title (:title item)
-   :inputprompt (:inputprompt item)
    :optional (:formitemoptional item)
    :type (:type item)
+   ;; TODO here we do a db call per item, for licenses we do one huge
+   ;; db call. Not sure which is better?
+   :localizations (into {} (for [{:keys [langcode title inputprompt] :as foo}
```
Comment (tjalkane) [inline src/clj/rems/db/applications.clj]: what is this `:as foo` doing here

File: src/clj/rems/db/applications.clj [hunk 2]
```diff
@@ -326,6 +332,16 @@
                    (and (or (nil? start) (time/before? start now))
                         (or (nil? end) (time/before? now end))))))))
 
+(defn- get-licenses-and-localizations [application catalogue-item-ids]
+  ;; TODO catalogue-item-ids passed in just for performance
+  (let [license-localizations (->> (db/get-license-localizations)
+                                   (map #(update-in % [:langcode] keyword))
+                                   (group-by :licid))
+        time (or (:start application) (time/now))]
```
Comment (tjalkane) [inline src/clj/rems/db/applications.clj]: isn't `time` redundant here?

File: src/clj/rems/db/applications.clj [hunk 3]
```diff
@@ -367,20 +384,16 @@
...
+                     (db/get-form-items {:id form-id}))
+         licenses (get-licenses-and-localizations application catalogue-item-ids)
          review-type (cond
                        (can-review? application) :normal
                        (can-third-party-review? application) :third-party
                        :else nil)]
      (when application-id
        (when-not (may-see-application? application)
          (throw-unauthorized)))
+     (prn :DBG items)
```
Comment (tjalkane) [inline src/clj/rems/db/applications.clj]: this should be removed

**Pull request merged**